Core/CodeGeneration/ThreadLocalAssemblyContextPoolDecorator.cs

```diff
@@ -59,7 +59,10 @@ public void DecrementCount ()
       }
     }
 
-    private readonly string _instanceID = Guid.NewGuid().ToString();
```

Comment:
okay, possible problem:

if we were to decide to use two separate assemblycontextpools on the same thread, this would have worked before but would mash them together now. I think we should keep the dictionary approach with an ID to separate instances of assemblycontextpools. 

probably, we could/should change the ID to an object, since it would provide better performance than a string.

For some extra performance, we might also consider using an array with 5 slots (and re-allocate a larger one as needed)  or simply go with a HybridDictionary, instead.

Reply:
That is not a problem since the `ThreadLocal` is not static. Different instances point to different thread local "slots". As such, having multiple instances of the decorator works correctly.

Reply:
That true...